### PR TITLE
SALTO-7092: call workspace close method in case loadWorkspace fails

### PR DESCRIPTION
**Pull request by dorsev**
Suggestion: View this PR using
image

---

_Release Notes_: 
None

---

_User Notifications_: 
None

### CODE REVIEW

File: packages/workspace/src/workspace/workspace.ts [hunk 1]
```diff
@@ -500,1095 +500,1116 @@ export const loadWorkspace = async (
...
-                        async staticFile =>
-                          (await naclFilesSource.getStaticFile({
-                            filePath: staticFile.filepath,
-                            encoding: staticFile.encoding,
-                            env: envName,
-                            isTemplate: staticFile.isTemplate,
-                            hash: staticFile.hash,
-                          })) ?? staticFile,
-                      ),
+  const close = async (): Promise<void> => {
```
Comment (yelly) [inline packages/workspace/src/workspace/workspace.ts:504]: This PR indents 1.1K lines, not great IMO.
A different approach can be renaming `loadWorkspace` to something else (`unsafeLoadWorksapce`, `loadWorkspaceInternal` etc.) and then implement `loadWorkspace` to just `try` calling the new function and closing in `except`.

Comment (dorsev) [reply packages/workspace/src/workspace/workspace.ts:504]: Your comment makes sense, but if I do as you suggest, I end up with the issue I sought to solve in the first place.

```
const loadWorkspace = {
try {
  const w = loadWorkspaceInternal()
} catch (e) {
// What do I call here? workspace is not 100% initialized.
}
```


So Im not sure how to address this comment at the moment. 

thoughts @ori-moisis @yelly ?

Comment (yelly) [reply packages/workspace/src/workspace/workspace.ts:504]: Yeah I guess the only real solution is breaking up this massive function but that's definitely out of scope.

**Review by yelly — APPROVED** [review]

File: packages/workspace/src/workspace/workspace.ts (hunk 1 above)
Comment (ori-moisis) [reply packages/workspace/src/workspace/workspace.ts:504]: tried to think of an alternative as well and could not come up with a good one

**Review by ori-moisis — APPROVED** [review]

**Comment by dorsev** [comment]
Thank you for the reviews @ori-moisis @yelly

**Pull request merged**